src/main/scala/chisel3/simulator/Simulator.scala

```diff
@@ -118,12 +141,69 @@ trait Simulator[T <: Backend] {
     val workspace = new Workspace(path = workspacePath, workingDirectoryPrefix = workingDirectoryPrefix)
     workspace.reset()
     val elaboratedModule =
-      workspace.elaborateGeneratedModule(
+      workspace
+        .elaborateGeneratedModule(
+          () => module,
+          args = chiselOptsModifications(chiselOpts).toSeq,
+          firtoolArgs = firtoolOptsModifications(firtoolOpts).toSeq
+        )
+    _simulate(workspace, elaboratedModule, settings)(body)
+  }
+
+  final def simulateTests[T <: RawModule with HasTests, U](
+    module:           => T,
+    includeTestGlobs: Seq[String],
+    chiselOpts:       Array[String] = Array.empty,
+    firtoolOpts:      Array[String] = Array.empty,
+    settings:         Settings[TestHarness[T]] = Settings.defaultRaw[TestHarness[T]]
+  )(body: (SimulatedModule[TestHarness[T]]) => U)(
+    implicit chiselOptsModifications: ChiselOptionsModifications,
+    firtoolOptsModifications:         FirtoolOptionsModifications,
+    commonSettingsModifications:      svsim.CommonSettingsModifications,
+    backendSettingsModifications:     svsim.BackendSettingsModifications
+  ): Seq[SimulatedTest] = {
```

Comment:
One big deviation from `simulate` is that `simulate` will assert if the test fails while `simulateTests` will not. This is going to be a big footgun for users as they need to remember to check that all their tests pass. Instead, we should aggressively error if any test fails.

Instead, aggregate all the failing tests into a single exception with a very verbose message that shows all the tests that are failing.